util.c

```diff
@@ -1658,6 +1662,9 @@ x_readlink(const char *path)
 }
 #endif
 
+/* Avoid breaking the api */
```

Comment:
I would prefer to update the signature of `read_file` instead of relying on a global.